src/cmake/AxomOptions.cmake

```diff
@@ -6,6 +6,7 @@
 # Defines CMake options for Axom's build system
 #------------------------------------------------------------------------------
 
+option(AXOM_ENABLE_ANNOTATIONS "Enables code annotations to facilitate performance evaluation." OFF)
```

Comment:
Are we testing this option in any of our CI plans?

If not, please update one of the TPLs to test this new feature.

Reply:
No, we are not -- I'll turn it ON for one of the CUDA builds.

Reply:
Done. See c755be7